state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,147 @@
+        gyro = pycubed.gyro
+        time.sleep(wait_time - random_wait_time)
+        acc = mid_acc
+        print("Data Collection Complete")
+        return acc, gyro, 0.0
+
+    elif testidx == "Rotating":
+        print("Please put cubesat on turntable. Waiting", wait_time, "seconds.")
+        time.sleep(wait_time)
+        print("Collecting IMU data...")
+
+        acc = pycubed.acceleration
+        gyro = pycubed.gyro
+        print("Data Collection Complete")
+        return acc, gyro, 0.0
+
+    elif testidx == "Magnet":
+        print("Please leave the cubesat flat on a table and retrieve a magnet. " +
+              "Waiting", wait_time, "seconds")
+        time.sleep(wait_time)
+        print("Please slowly move the magnet closer to the cubesat for", mag_time, "seconds")
+
+        acc = pycubed.acceleration
+        gyro = pycubed.gyro
+        starting_mag = pycubed.magnetic
+        time.sleep(mag_time)
+        ending_mag = pycubed.magnetic
+        # calculate the change in the total magnetometer reading
+        mag = (sqrt(ending_mag[0] ** 2 + ending_mag[1] ** 2 + ending_mag[2] ** 2) -
+               sqrt(starting_mag[0] ** 2 + starting_mag[1] ** 2 + starting_mag[2] ** 2))
+        print("Data Collection Complete")
+        return acc, gyro, mag
+
+
+def imu_test(cubesat, result_dict, testidx):
+    """
+    All automation happens in this function
+    Select burnwire and voltage level
+    Process user test results and update the result dictionary accordingly
+    """
+    if testidx == "Still":
+        # record acceleration
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = "Testing IMU when cubesat is still. Acc: " + str(acc)
+
+        # if total acceleration is within 0.2 m/s^2 of 9.8, true
+        if abs(9.8 - sqrt(acc[0] ** 2 + acc[1] ** 2 + acc[2] ** 2)) < 0.2:
+            result_dict['IMU_Acc_Still'] = (result_val_string, True)
+        else:  # else false
+            result_dict['IMU_Acc_Still'] = (result_val_string, False)
+
+    elif testidx == "Moving":
+        # record acceleration
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = "Testing IMU when cubesat is being moved. Acc: " + str(acc)
+
+        # if each acceleration component is more than 0, true
+        if abs(acc[0]) > 0 and abs(acc[1]) > 0 and abs(acc[2]) > 0:
+            result_dict['IMU_Acc_Moving'] = (result_val_string, True)
+        else:  # else false
+            result_dict['IMU_Acc_Moving'] = (result_val_string, False)
+
+    elif testidx == "Rotating":
+        # record gyro reading
+        acc, gyro, mag = user_test(cubesat, result_dict, testidx)
+        result_val_string = "Testing IMU when cubesat is rotating on a turntable. Gyro: " + str(gyro)
+
+        # if total gyro reading is within 2 deg/s^2 of angular velocity, return true
+        ang_vel = float(input("Please enter the speed the turntable was rotating at in deg/s: "))
```

Comment:
Probably unrealistic for someone to know this number.

Reply:
Just have the person rotate it in their hand, and check that it has changed from when the cubesat was still.

Reply:
Eventually we'll use a turntable, but that won't fall under the umbrella of this script.